### PR TITLE
Issue #175 - implement array with indicator indentation flag

### PR DESCRIPTION
**Pull request by dswiecki**
Solves #175

### CODE REVIEW

**Comment by cowtowncoder** [comment]
Sounds like a useful addition (haven't reviewed change in detail but makes sense).
Just one quick note: since it is an API addition, it must go in a new minor version; in this case 2.12.0.
Timing works well as I am about to release first release candidate.

So, branch to merge would need to be 2.12.

File: yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java [hunk 1]
```diff
@@ -116,6 +116,15 @@
          * @since 2.9
          */
         INDENT_ARRAYS(false),
+        /**
+         * Feature enabling of which adds indentation with indicator for array entry generation
+         * (default indentation being 2 spaces).
+         *<p>
+         * Default value is `false` for backwards compatibility
+         *
+         * @since 2.11.3
```
Comment (cowtowncoder) [inline yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java:125]: As per not on PR itself, new features need to be added in a minor version, not patch (as per SemVer guidance)

**Comment by cowtowncoder** [comment]
Ok; ready to merge once I get CLA (see notes on issue), and PR gets rebased against 2.12 (to add a new feature).

Thank you once again for contributing this!

File: yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java (hunk 1 above)
Comment (dswiecki) [reply yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java]: updated to 2.12.0

**Comment by dswiecki** [comment]
Updated PR, unfortuanetely can't rename branch - it has to be named 2.11 but it's based on 2.12. I've also sent CLA.

**Comment by cowtowncoder** [comment]
@dswiecki np wrt name, only matters what it merges against. Will have a quick re-read; received CLA.

**Comment by cowtowncoder** [comment]
@dswiecki Works great for 2.12, although there is a problem with 3.0 (`master`) -- looks like the new snakeyaml-engine does not have method `setIndentWithIndicator` in `DumpSettingsBuilder`, and I am not sure what the equivalent might be.
I was wondering if you know what might be the replacement there? If not, we can ask @asomov for help too.

**Comment by asomov** [comment]
Indeed, this is not yet ported to snakeyaml-engine. I will do as soon as I can. Feel free to create an issue for snakeyaml-engine: https://bitbucket.org/asomov/snakeyaml-engine/issues?status=new&status=open

**Comment by cowtowncoder** [comment]
@asomov created https://bitbucket.org/asomov/snakeyaml-engine/issues/27/support-snakeyaml

**Comment by asomov** [comment]
@cowtowncoder delivered in version 2.2
https://search.maven.org/artifact/org.snakeyaml/snakeyaml-engine/2.2/bundle

**Comment by cowtowncoder** [comment]
@asomov Excellent, update `master` to use 2.2, option added!

**Comment by asomov** [comment]
@cowtowncoder I am terribly sorry. Even though we have high test coverage a regression happen in SnakeYAML Engine version 2.2
New lines in the Windows format (\r\n) are not always respected properly.
There is already a fix in version 2.2.1
Can you please use the fixed version of SnakeYAML Engine ? Should I create a PR for the version change ?

**Comment by cowtowncoder** [comment]
@asomov Sure PR would be good if you  have time.

**Comment by asomov** [comment]
@cowtowncoder done: https://github.com/FasterXML/jackson-dataformats-text/pull/234

**Pull request merged**